src/mapfile2js/src/parse/parseBlockKey.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Parse block keys.
+ * @param {Object} lo Line Object
+ * @param {number} index Current lines index
+ * @param {array} lines Array of line strings
+ * @param {array} blocks Block stack
+ */
+
+// there can be multiple layer, class and style block siblings
+const multi_block_keys = ["layer", "class", "style"];
+
+function parseBlockKey(lo, index, lines, blocks) {
```

Comment:
Can we avoid side effects on the blocks variable?

Reply:
yes, done in #2